### ISSUE TITLE
release find by tag returns 500 error

### ISSUE THREAD

**Issue by cameronbraid** [go-gitea/gitea#14365]
git server version 1.13.1

When using the `Get a release by tag name` api call a 500 error is returned for a tag_name after `git tag` has been done but a release has not been created.  The response is

{
  "message": "user does not exist [uid: 0, name: , keyid: 0]",
  "url": "https://git-dev.drivenow.com.au/api/swagger"
}

I suspect the cause is because there exists a record in the release table with the publisher_id=0

**Comment by noerw** [comment]
Can't reproduce this on try.gitea.io.
Please provide a way to reproduce the issue, otherwise we can't debug this.

**Comment by zeripath** [comment]
Or at least give us some logs from the gitea instance.

**Comment by cameronbraid** [comment]
I am using sqlite as database

the logs dont show anything

```
[36m2021/01/17 13:00:38 [0mCompleted [34mGET[0m [1m/api/v1/repos/drivenow/frontend/releases/tags/v8.0.13[0m [1;41m500[0m [1;41mInternal Server Error[0m in [1m62.717056ms[0m
```

**Comment by cameronbraid** [comment]
Looks like try uses master as it has this code in there :

```
func (r *Release) loadAttributes(e Engine) error {
	var err error
	if r.Repo == nil {
		r.Repo, err = GetRepositoryByID(r.RepoID)
		if err != nil {
			return err
		}
	}
	if r.Publisher == nil {
		r.Publisher, err = getUserByID(e, r.PublisherID)
		if err != nil {
			if IsErrUserNotExist(err) {
				r.Publisher = NewGhostUser()
			} else {
				return err
			}
		}
	}
	return getReleaseAttachments(e, r)
}

```

where as v1.13.1 is missing the `IsErrUserNotExist` test

**Comment by noerw** [comment]
Thanks, good hint!
Looks like #12096 did a bugfix within a feature PR :| Will backport that to 1.13.2

**Comment by 6543** [comment]
closed by #14375
